### PR TITLE
Allow Hapi server to be customized during initialization

### PR DESCRIPTION
**Pull request by pascalw**
With this, the Hapi server can be customized during initialization.
Use cases are outlined in https://github.com/pascalw/dashbling/issues/20.

### CODE REVIEW

File: example/dashbling.config.js [hunk 1]
```diff
@@ -20,6 +20,22 @@ module.exports = {
...
+      method: "GET",
+      path: "/ping",
+      handler: (_request, _h) => {
+        return "pong";
+      }
+    });
+
+    // Be sure to return a Promise, so the initialization process
+    // waits for this configuration to be completed.
+    Promise.resolve();
```
Comment (eshaham) [inline example/dashbling.config.js]: @pascalw I know this is just an example, but should it be `return Promise.resolve()`?

Comment (pascalw) [reply example/dashbling.config.js]: Good catch! I've been doing to much ReasonML lately haha 🙂

**Comment by pascalw** [comment]
@eshaham I've published this change as `0.2.0` on NPM, under a `beta` dist tag. You can try it by changing `@dashbling/core` version in `package.json` to `0.2.0`.

File: example/dashbling.config.js (hunk 1 above)
Comment (pascalw) [reply example/dashbling.config.js]: I've updated the example to use an async function, so you don't have to explicitly return a promise.

Comment (eshaham) [reply example/dashbling.config.js]: 👍

**Pull request merged**